File: Javascript/Level 3/Problem 1.js
function calculateAverage(arr) {
    let total = 0;
    // BUG: Loop condition uses '<=' causing an out-of-bound error.
    for (let i = 0; i <= arr.length; i++) {  
      total += arr[i];
    }
    return total / arr.length;
  }
  console.log(calculateAverage([10, 20, 30, 40, 50]));
  

File: Python/Level 3/Problem 1.py
def calculate_average(arr):
    total = 0
    # BUG: Loop iterates one extra time, causing an IndexError.
    for i in range(len(arr) + 1):
        total += arr[i]
    return total / len(arr)

print(calculate_average([10, 20, 30, 40, 50]))
